### ISSUE TITLE
Multiclass classification

### ISSUE THREAD

**Issue by lishubin** [dmlc/xgboost#216]
Hello, thank you for the amazing package. I am using the multi:softmax and multi:softprob. it requires me to input the num_class. it seems to be the parameter is always 1 more the class i have in the training set. therefore, when i use the softprob, it returns 1 more class than i have in my training set, i need some help interpret this "additional" class. thank you

**Comment by tqchen** [comment]
by convention, xgboost index starts from 0, which means the classes index should be in [0,num_class -1]

**Comment by lishubin** [comment]
Thank you. I created a dataset with my Y value starting from 0 to 17. when I use softmax, I need to specify num_class = 19, and the predict output is from 1 to 18. when I use softprob in the same case, i have 19 predict probabilities for each instance. If i use num_class = 18, xgboost always crash. I am wondering how to interpret these 19 probabilities while in my training set there is only 18 class. hope this makes my problem more clear.

**Comment by tqchen** [comment]
I have updated the repo to avoid crash and have better error message, could you try the newest version?  If you still found problem, it would be great if you can post a small piece of code to reproduce the error(can be based on randomly generated data)

**Comment by tqchen** [comment]
any updates? I will close this for now, please re-open if you have updates
